[LifHoshi] iP

LifHoshi opened the pull request:
### **PingPro**

> “_Your mind is for having ideas, not holding them_.”  – David Allen ([source](https://dansilvestre.com/productivity-quotes))

PingPro frees your mind of having to remember things you need to do. It's,
• text-based
• easy to learn
• FAST SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2.	double-click it.
3.	add your tasks.
4.	let it manage your tasks for you 😉

And it is **FREE!**

**_Features:_**

- [x]  Managing tasks

- [ ]  Managing deadlines (coming soon)

- [ ]  Reminders (coming soon)

________________________________________
If you are a Java programmer, you can use it to practice Java too. Here's the ```main```  method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

LifHoshi committed:
Rename,Greet, Exit

LifHoshi committed:
Updated Rename, Greet, Exit

LifHoshi committed:
Echo

LifHoshi committed:
Add, List

LifHoshi committed:
Mark as Done

LifHoshi committed:
ToDos, Events, Deadlines

LifHoshi committed:
A-TextUiTesting

LifHoshi committed:
Handle Errors

LifHoshi committed:
Delete

LifHoshi committed:
Level-7 Add TaskStorage

LifHoshi committed:
Level-8 Changed the String date to correct formate of date

LifHoshi committed:
Added A-MoreOOP

LifHoshi committed:
Added Package, add Package done with OOP process at the same time

LifHoshi committed:
Merge remote-tracking branch 'origin/add-gradle-support'

LifHoshi committed:
Added A-Gradle

LifHoshi committed:
Added A-Gradle

LifHoshi committed:
Updated  A-Gradle

LifHoshi committed:
Added A-JUnit

LifHoshi committed:
Added  A-Jar file

LifHoshi committed:
Added JavaDoc in each file

LifHoshi committed:
Tweaked the code to comply with a coding standard already.

LifHoshi committed:
Add related file about Level-9 find

LifHoshi committed:
Follow the Gradle CheckStyle

kjw142857 commented on src/main/java/Command/UnMarkCommand.java:
```diff
@@ -0,0 +1,28 @@
+/**
+ * This class is used to unmark a task
+ */
+public class UnMarkCommand extends Command {
```
Here, "Unmark" seems to be one word to me! Maybe you could rename this class as `UnmarkCommand`?

kjw142857 commented on src/main/java/Ping/Storage.java:
```diff
@@ -0,0 +1,65 @@
+package Ping;
+import java.io.*;
```
Here, the coding standard needs imports to be stated explicitly. Perhaps you could list out all the imports, such as `java.io.File`?

kjw142857 commented on src/test/java/Ping/DeadlineTest.java:
```diff
@@ -0,0 +1,23 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class DeadlineTest {
+    @Test
+    public void todoCommand_ShouldHaveCorrectForamt() {
```
```suggestion
    public void todoCommand_ShouldHaveCorrectFormat() {

```
Here I think there may be a spelling error, did you mean `Format` instead of `Foramt`?

kjw142857 reviewed (commented):
Overall, good job - looks like you have used OOP principles well and your names are largely sensible! I saw a few typos here and there in your naming, but nothing too major - maybe you would want to proofread your code again to catch any spelling/naming errors.

LifHoshi committed:
Implement GUI

LifHoshi committed:
Implement Varargs in FindCommand

jerremyng commented on src/main/java/Command/AddCommand.java:
```diff
@@ -0,0 +1,27 @@
+ * This class is used to add a task to the list of tasks
+ */
+public class AddCommand extends Command {
+    private Task task;
+    public AddCommand(Task task) {
```
A linebreak here might make it more readable to distinguish between these two members
```suggestion
    private Task task;
    
    public AddCommand(Task task) {

```

jerremyng commented on src/main/java/Command/AddCommand.java:
```diff
@@ -0,0 +1,27 @@
+        if (task == null) {
+            throw new PingException("Task cannot be empty");
+        } else {
+        tasks.addTask(task);
+        return ui.addMessage(task, tasks.taskSize());
```
Small indentation errors here, happens to the best of us!
```suggestion
            tasks.addTask(task);
            return ui.addMessage(task, tasks.taskSize());

```

jerremyng commented on src/main/java/Ping/Parser.java:
```diff
@@ -0,0 +1,196 @@
+        case "bye":
+            return new ExitCommand();
+        case "mark":
+            return parseMark(restCommands);
+        case "unmark":
+            return parseUnmark(restCommands);
+        case "todo":
+            return parseTodo(restCommands);
+        case "list":
+            return new ListCommand();
+        case "delete":
+            return parseDelete(restCommands);
+        case "deadline":
+            return parseDeadline(restCommands);
+        case "event":
+            return parseEvent(restCommands);
+        case "find":
+            return parseFind(restCommands);
```
I like how you seperated the parse methods into private methods in this class, makes it cleaner instead of one long method!

jerremyng commented on src/main/java/Ping/Parser.java:
```diff
@@ -0,0 +1,196 @@
+
+    private static Command parseEvent(String[] evCommand) throws PingException {
+        StringBuilder rest = new StringBuilder();
+        StringBuilder date1 = new StringBuilder();
+        StringBuilder date2 = new StringBuilder();
```
Perhaps these variables can be renamed to be more meaningful according to the naming standards.
```suggestion
        StringBuilder startDate = new StringBuilder();
        StringBuilder endDate = new StringBuilder();
```

jerremyng reviewed (commented):
LGTM, great work on your project structure, seems clean to me!

LifHoshi committed:
Parser class: Parser the commend of the user input

Parser class have some unintended and potential mistake could not be
noticed due to user invalid input. (e.g. mark number out of list).

Adding assert code on every necessarily needs part preventing from this
issue. For example, assert rest commands length to check whether it
matches the required input.

Although previous code had already handled this issue, this week's iP
project required to do so.

Refer to this website on week 5 to see more:
https://nus-cs2103-ay2324s2.github.io/website

LifHoshi committed:
Parser class: Parser the commend of the user input.

Parser class have some unintended and potential mistake could not be
noticed due to user invalid input. (e.g. mark number out of list).

Adding assert code on every necessarily needs part preventing from this
issue. For example, assert rest commands length to check whether it
matches the required input.

Although previous code had already handled this issue, this week's iP
project required to do so.

Refer to this website on week 5 to see more:
https://nus-cs2103-ay2324s2.github.io/website

LifHoshi committed:
Parser class: Parser the commend of the user input

Parser class have some unintended and potential mistake could not be
noticed due to user invalid input. (e.g. mark number out of list).

Adding assert code on every necessarily needs part preventing from this
issue. For example, assert rest commands length to check whether it
matches the required input.

Although previous code had already handled this issue, this week's iP
project required to do so.

Refer to this website on week 5 to see more:
https://nus-cs2103-ay2324s2.github.io/website

LifHoshi committed:
Merge branch 'A-CodeQuality'

LifHoshi committed:
Merge branch 'master' into A-Assertions

LifHoshi committed:
Parser class: Parser the commend of the user input
Parser class has some code quality issues after learning this week
content, like Magic number, Complicated expression, etc.

Modifying the some lines of code to make it more readable, tidy,
followed the code quality requirement.

Doing this way just follow the code quality requirement. And a good
chance to learn

Refer to this website to seek more:
https://nus-cs2103-ay2324s2.github.io/website

LifHoshi committed:
Merge branch 'A-CodeQuality'

LifHoshi committed:
Merge pull request #1 from LifHoshi/branch-A-Assertions

Branch a assertions

LifHoshi committed:
Change lines for PR requirement

LifHoshi committed:
Merge pull request #2 from LifHoshi/branch-A-CodeQuality

Change lines for PR requirement

LifHoshi committed:
Parser class: parser the command into already exist command

There is a import order error of Parser class. Change to correct order
now.

TaskList class: handle the task in a list.

Add a new detectDuplicate method to check whether there is same task
exist in current list.

AddCommand class: a class that adding task into list

*Update with a new else-if condition to check duplicate task

LifHoshi committed:
Merge pull request #3 from LifHoshi/C-DetectDuplicates

Parser class: parser the command into already exist command

LifHoshi committed:
Modify most of classes to have a hierarchy structure

For example, the original packages only have two which is Ping and
Command. Renamed them with lower case to fit the standard of commit
then separate all of them into 4 different packages which are
command, exceptions, GUI, job, and a package folder for checking
time.

Implement a better design GUI to make the window more beautiful and
tidy.

A good and comfortable window will improve the user experience. So,
doing a GUI beauty improvement is necessary.

Let's
* change the font to make chatbot and user different
* add different background colors to distinguish chatbot and user

LifHoshi committed:
Merge pull request #5 from LifHoshi/A-BetterGui

Modify most of classes to have a hierarchy structure

LifHoshi committed:
Change the files of Deadline, Todo to make it more OOP

The original implement a method but not override in the rest of
children, yet still implement again. If the father of the class own
the method than the child shouldn't write again.

Upload a Ui photo.

LifHoshi committed:
Update README.md

LifHoshi committed:
Add files via upload

LifHoshi committed:
Add files via upload

LifHoshi committed:
Update README.md

LifHoshi committed:
Update README.md

LifHoshi committed:
Add files via upload

LifHoshi committed:
Update README.md

LifHoshi committed:
Merge pull request #6 from LifHoshi/LifHoshi-patch-1

Update the user guide

LifHoshi committed:
Implement a better GUI and add some notifications

A better GUI implement but forgot to add a tag. In addition, add some
notifications about where parts of codes comes from. Showing reference.
